### PR TITLE
Implement questionnaire-unit and questionnaire-unitOption extensions

### LINKED ISSUE #1550: Fix support for Quantity-type questions and answers
**Is your feature request related to a problem? Please describe.**
The [full definition](https://www.hl7.org/fhir/codesystem-item-type.html#item-type-quantity) of the "quantity" type is:
> Question with a combination of a numeric value and unit, potentially with a comparator (<, >, etc.) as an answer. (valueQuantity) There is an extension 'http://hl7.org/fhir/StructureDefinition/questionnaire-unit' that can be used to define what unit should be captured (or the unit that has a ucum conversion from the provided unit).

`http://hl7.org/fhir/StructureDefinition/questionnaire-unit` extension support is not listed either yes or no on [FHIR specification support page](https://github.com/google/android-fhir/wiki/SDCL%3A-FHIR-specification-support), but based on some testing, we seem to ignore it. However, in the definition of the `questionnaire-unit` extension it states:
> Context of Use: Use on Element ID Questionnaire.item or Element ID Questionnaire.item.item, with [Context Invariant](http://hl7.org/fhir/R4B/structuredefinition-definitions.html#StructureDefinition.contextInvariant) = `type='integer' or type='decimal'`

So the actual way to use `type='quantity'` are the [`unitOption` or `unitValueSet` extensions](http://hl7.org/fhir/uv/sdc/behavior.html#choice-restriction) which we explicitly do not support, but without which `type='quantity'` is just a strictly worse version of `'integer'` or `'decimal'`.

You can see examples of both versions in this [LHC Forms unit test questionnaire JSON](https://github.com/lhncbc/lforms-fhir-app/blob/master/e2e-tests/data/R4/vital-sign-questionnaire.json).

**Describe the solution you'd like**
Add support for the `questionnaire-unit` extension. If it is used on an item of `type='integer'` or `type='decimal`, it should:
1. Visibly show the unit in the rendered Questionnaire
2. In the questionnareResponse, generate a [valueQuantity](https://www.hl7.org/fhir/datatypes.html#Quantity) 
3. Populate the valueQuantity with any `unit`, `system`, `code` information provided in `questionnaire-unit`

Prioritize support for `unitOption` or `unitValueSet`, where it should:
1. Show a unit picker (dropdown?) in the rendered Questionnaire
2. In the questionnareResponse, generate a [valueQuantity](https://www.hl7.org/fhir/datatypes.html#Quantity) 
3. Populate the valueQuantity with any `unit`, `system`, `code` information provided in `questionnaire-unit`

Lower/caveat the stated support of `type='quantity'` until a method of unit selection is available.

### LINKED ISSUE #1600: Add support for unitOption and/or unitValueSet
This is a continuation of #1550, see there for additional context.

A primary purpose of Questionnaire item `type='quantity'` is the ability for the user to select a unit from a list of options using the [`unitOption` or `unitValueSet` extensions](http://hl7.org/fhir/uv/sdc/behavior.html#choice-restriction). We currently do not support either, so `type='quantity'` items are not very usable in SDC.

**Describe the solution you'd like**
Add support for `unitOption` or `unitValueSet`, where it should:
1. Show a unit picker (dropdown?) in the rendered Questionnaire
2. In the questionnareResponse, generate a [valueQuantity](https://www.hl7.org/fhir/datatypes.html#Quantity)
3. Populate the valueQuantity with any `unit`, `system`, `code` information provided in `questionnaire-unit`

### PR DESCRIPTION
**Pull request by jingtang10**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #1550 
Fixes #1600 

**Description**
Clear and concise code change description. 

**Alternative(s) considered**
Have you considered any alternatives? And if so, why have you chosen the approach in this PR?

**Type**
Choose one: (Bug fix | Feature | Documentation | Testing | Code health | Builds | Releases | Other)

**Screenshots (if applicable)**

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

**Comment by santosh-pingle** [comment]
@jingtang10 
Can you please upload the screenshots?

**Review by santosh-pingle — COMMENTED** [review]
Can you please write unit tests to check whether units are shown as per the `questionnaire-unitOption`, `questionnaire-unit`?

**Comment by santosh-pingle** [comment]
Please declare `@dimen/padding_default` to dimens.xml, at the moment change is not committed.

**Comment by santosh-pingle** [comment]
https://github.com/google/android-fhir/issues/2013

File: catalog/src/main/assets/component_quantity_with_validation.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{
+  "resourceType": "Questionnaire",
+  "item": [
+    {
+      "linkId": "1",
+      "text": "How many times have you been pregnant?",
+      "type": "quantity",
```
[the remark this answers is not in the document]
Comment (jingtang10) [reply catalog/src/main/assets/component_quantity_with_validation.json]: done. this is an old file and i forgot to update. updated. thanks!

**Comment by jingtang10** [comment]
> @jingtang10 
> Can you please upload the screenshots?

**Comment by jingtang10** [comment]
> Please declare `@dimen/padding_default` to dimens.xml.

is this comment still valid? sorry i'm not sure what to do here. thanks!

**Comment by jingtang10** [comment]
> Can you please write unit tests to check whether units are shown as per the `questionnaire-unitOption`, `questionnaire-unit`?

done

**Review by santosh-pingle — APPROVED** [review]
@jingtang10 thanks!

**Pull request merged**